### PR TITLE
feat: new quick filters UI and functionality

### PR DESCRIPTION
**Pull request by egoroza**
New (2/27)
Screen Shot 2023-02-27 at 1 24 34 PM
Screen Shot 2023-02-27 at 1 25 30 PM
Screen Shot 2023-02-27 at 1 25 38 PM

OLD:
Screen Shot 2023-02-21 at 3 51 28 PM
Screen Shot 2023-02-21 at 3 52 14 PM

### CODE REVIEW

File: src/components/LoansByCategory/QuickFilters/FilterPills.vue [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+					this.selectedOptions = this.selectedOptions.filter(val => val !== option);
+				}
+			}
+			this.$emit('update-values', { values: [...this.selectedOptions], selected, option });
+		}
+	}
+};
+</script>
+
+<style lang="postcss" scoped>
```
Comment (mcstover) [inline src/components/LoansByCategory/QuickFilters/FilterPills.vue:70]: Could we please convert as much of these styles as possible to use tailwind variables.

File: src/components/LoansByCategory/QuickFilters/LocationSelector.vue [hunk 2]
```diff
@@ -309,7 +356,29 @@ export default {
 };
 </script>
 
-<style scoped>
+<style lang="postcss" scoped>
+.filter-pill {
```
Comment (mcstover) [inline src/components/LoansByCategory/QuickFilters/LocationSelector.vue:360]: Same here, we need to use our styleguide vars for this.

File: src/pages/Lend/LoanChannelCategoryControl.vue [hunk 3]
```diff
@@ -375,7 +375,7 @@ export default {
 			flssLoanSearch: {},
 			quickFiltersOptions: {
 				gender: [{
-					key: '',
+					key: 'all',
```
Comment (dyersituations) [inline src/pages/Lend/LoanChannelCategoryControl.vue:378]: Just out of curiosity, why the change to `all` instead of empty string?

File: src/components/LoansByCategory/QuickFilters/QuickFilters.vue [hunk 4]
```diff
@@ -345,6 +369,29 @@ export default {
 </script>
 
 <style lang="postcss" scoped>
+	.filter-pill {
```
Comment (dyersituations) [inline src/components/LoansByCategory/QuickFilters/QuickFilters.vue:372]: Some of the CSS here can be changed to tailwind as well, if not in the template (since we don't have all of the versions like hover in our compiled TW - we don't currently rely on JIT tailwind), here with `@apply`.

File: src/components/LoansByCategory/QuickFilters/QuickFilters.vue [hunk 5]
```diff
@@ -179,13 +184,16 @@ export default {
...
+		KvMaterialIcon,
+		FilterPills
 	},
 	data() {
 		return {
 			mdiFilterVariant,
+			mdiSort,
 			selectedCategory: 0,
 			selectedGender: '',
+			selectedGenders: ['all'], // TODO: consolidate to just this when UI changes apply to QF on lending home, too
```
Comment (dyersituations) [inline src/components/LoansByCategory/QuickFilters/QuickFilters.vue:196]: Not sure what this comment means. Please add a ticket and reference here with details on what this `TODO` entails.

File: src/components/LoansByCategory/QuickFilters/QuickFilters.vue [hunk 6]
```diff
@@ -298,10 +310,21 @@ export default {
 				location
 			);
 		},
+		// We will need to factor in categories at some point
```
Comment (dyersituations) [inline src/components/LoansByCategory/QuickFilters/QuickFilters.vue:313]: Not sure what this comment means. Consider adding a ticket and referencing here, if future changes are needed.

**Comment by dyersituations** [comment]
In desktop view, I'm seeing a white box/border under + around the gender pills. This box doesn't appear when using mobile mode of chrome dev tools.
image

**Comment by dyersituations** [comment]
Clicking either the location or sort by "activates" the textbox. The active/focus border can be removed.
image

**Comment by dyersituations** [comment]
Clicking outside the textbox on the sort by filter doesn't open the dropdown. The clickable area is vaguely shown by this red area:

image

**Comment by dyersituations** [comment]
The down arrows on the pills are slightly different sizes and misaligned

image

**Comment by dyersituations** [comment]
The box shadow on the gender filter pills is cut off. It's subtle, but there's a difference between the gender filter shadows and the other filters
image

**Comment by dyersituations** [comment]
Left padding missing in mobile view (notice difference between left and right padding).
image

**Comment by dyersituations** [comment]
There's a small height different in filter pill heights

image

**Comment by dyersituations** [comment]
LH filters now show an empty gender filter, and the filters no longer align.

image

**Comment by dyersituations** [comment]
All genders in LH doesn't work.

image

File: src/pages/Lend/LoanChannelCategoryControl.vue (hunk 3 above)
Comment (egoroza) [reply src/pages/Lend/LoanChannelCategoryControl.vue:378]: Looks like if we have an empty string it won't apply it as an `id` property for html elements. Since the filter pills are technically checkbox inputs (`<input>`) when we pass in `:id=option.key` and `option.key === ''` it doesn't apply an ID at all and seems to ignore empty string, so any interactions with "All genders" doesn't get considered as part of the checkbox list state

Comment (egoroza) [reply src/pages/Lend/LoanChannelCategoryControl.vue:378]: Obviously could have worked around this but I thought it made sense to just change the `key` across the board instead of interpolating a random char/phrase or something similar

File: src/components/LoansByCategory/QuickFilters/QuickFilters.vue (hunk 5 above)
Comment (egoroza) [reply src/components/LoansByCategory/QuickFilters/QuickFilters.vue:196]: ya will do, just meant mostly that i don't know if it makes sense to have 2 gender states. The dropdown on lending home for gender should just use the array as well, there will just only ever be 1 element in the array at a time

File: src/pages/Lend/LoanChannelCategoryControl.vue (hunk 3 above)
Comment (dyersituations) [reply src/pages/Lend/LoanChannelCategoryControl.vue:378]: Sounds good, thanks for the info, using `all` is fine.

File: src/components/LoansByCategory/QuickFilters/QuickFilters.vue (hunk 5 above)
Comment (dyersituations) [reply src/components/LoansByCategory/QuickFilters/QuickFilters.vue:196]: In that case, maybe update the comment to mention that the work on LH might mean removing the other gender state, I'm assuming `selectedGender`?

**Comment by egoroza** [comment]
> In desktop view, I'm seeing a white box/border under + around the gender pills. This box doesn't appear when using mobile mode of chrome dev tools. image

How can one reproduce? Browser? I don't see this issue on my Chrome desktop view so need to investigate

**Comment by dyersituations** [comment]
> How can one reproduce? Browser? I don't see this issue on my Chrome desktop view so need to investigate

Interesting. I see this in Chrome on my MacBook on both my MacBook screen and external monitor. Make sure that the devtool mobile emulator isn't enabled, since it doesn't occur in the emulator.

File: src/components/LoansByCategory/QuickFilters/FilterPills.vue (hunk 1 above)
Comment (mcstover) [reply src/components/LoansByCategory/QuickFilters/FilterPills.vue:74]: Thanks for reducing the level of non-tailwind styles!

File: src/components/LoansByCategory/QuickFilters/QuickFilters.vue (hunk 5 above)
Comment (dyersituations) [reply src/components/LoansByCategory/QuickFilters/QuickFilters.vue:230]: Looks like this comment could still use some clarity, maybe mention that `selectedGender` could be removed, if that's the change needed here.

**Comment by dyersituations** [comment]
At a width slightly larger than small, the entire page gets horizontal scrolling. At a smaller view, the second row of filters scroll, but not at the width in this screenshot. Maybe the filters need horizontal scrolling at the medium screen breakpoint?

image

**Comment by dyersituations** [comment]
In Chrome without mobile emulation, I'm still seeing a weird box around the gender filters. That box is also cutting off the gender filter drop shadow and making the sort filter pill taller.

image

**Comment by dyersituations** [comment]
For consistency with the other filter pills, the sort filter should have the pointer icon for the entire pill. The pointer icon currently only shows up for the sort icon on hover.

File: src/components/LoansByCategory/QuickFilters/QuickFilters.vue [hunk 7]
```diff
@@ -40,7 +30,16 @@
 					</option>
 				</kv-select>
 			</div>
-			<div v-if="!removeGenderDropdown" class="tw-flex tw-flex-col tw-grow">
+			<div v-if="!removeGenderDropdown && !withCategories" class="tw-overflow-scroll tw-pb-1">
```
Comment (dyersituations) [inline src/components/LoansByCategory/QuickFilters/QuickFilters.vue:33]: This should probably be `tw-overflow-x-auto`. The box I'm seeing in the screenshot in another comment is the scrollbars always being visible.

**Comment by dyersituations** [comment]
Please create a follow-up ticket to address the cut-off drop-shadows. It's subtle, so I'll approve as is, but we should address later when working on the filters:

image

image

**Review by dyersituations — APPROVED** [review]
